experiment/invitations-accepter/main_test.go

```diff
@@ -0,0 +1,117 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"sort"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/sirupsen/logrus/hooks/test"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/github/fakegithub"
+)
+
+func TestAcceptInvitations(t *testing.T) {
+
+	testCases := []struct {
+		id           string
+		invitations  map[int]github.UserRepoInvitation
+		expected     []github.UserRepoInvitation
+		expectedLogs []*logrus.Entry
+	}{
+		{
+			id: "no invitations to accept",
+			expectedLogs: []*logrus.Entry{
+				{
+					Data:    logrus.Fields{"bot-user": "k8s-ci-robot"},
+					Message: "No pending invitations have been found.",
+					Level:   logrus.InfoLevel,
+				},
+			},
+		},
+		{
+			id: "one invitation to accept",
+			invitations: map[int]github.UserRepoInvitation{
+				1: {InvitationID: 1, Repository: &github.Repo{FullName: "foo/bar"}},
+			},
+			expectedLogs: []*logrus.Entry{
+				{
+					Data:    logrus.Fields{"bot-user": "k8s-ci-robot", "invitation-id": 1, "repo": "foo/bar"},
+					Message: "Accepting invitation.",
+					Level:   logrus.InfoLevel,
+				},
+			},
+		},
+		{
+			id: "multiple invitations per client to accept",
+			invitations: map[int]github.UserRepoInvitation{
+				1: {InvitationID: 1, Repository: &github.Repo{FullName: "foo/bar"}},
+				2: {InvitationID: 2, Repository: &github.Repo{FullName: "james/bond"}},
+				3: {InvitationID: 3, Repository: &github.Repo{FullName: "captain/hook"}},
+			},
+			expectedLogs: []*logrus.Entry{
+				{
+					Data:    logrus.Fields{"bot-user": "k8s-ci-robot", "invitation-id": 1, "repo": "foo/bar"},
+					Message: "Accepting invitation.",
+					Level:   logrus.InfoLevel,
+				},
+				{
+					Data:    logrus.Fields{"bot-user": "k8s-ci-robot", "invitation-id": 2, "repo": "james/bond"},
+					Message: "Accepting invitation.",
+					Level:   logrus.InfoLevel,
+				},
+				{
+					Data:    logrus.Fields{"bot-user": "k8s-ci-robot", "invitation-id": 3, "repo": "captain/hook"},
+					Message: "Accepting invitation.",
+					Level:   logrus.InfoLevel,
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.id, func(t *testing.T) {
+			hook := test.NewGlobal()
+
+			var fgh githubClient
+			fgh = &fakegithub.FakeClient{UserRepoInvitations: tc.invitations}
+
+			if err := acceptInvitations(fgh, false); err != nil {
+				t.Fatalf("error wasn't expected: %v", err)
+			}
+
+			hookEntries := hook.AllEntries()
+			sort.Slice(hookEntries, func(p, q int) bool {
+				return hookEntries[p].Message < hookEntries[q].Message
+			})
+
+			if diff := cmp.Diff(hookEntries, tc.expectedLogs, cmp.AllowUnexported(logrus.Entry{}), cmpopts.IgnoreFields(logrus.Entry{}, "Time", "Logger")); diff != "" {
+				t.Fatal(diff)
+			}
+
+			actualInvitations, _ := fgh.ListCurrentUserRepoInvitations()
```

Comment:
Please do not swallow errors, it might be changed in the future to return an error and that will be annoying to debug